### PR TITLE
Stream History Rollback

### LINKED ISSUE #1012: Add the option to rollback to a previous stream-version
As a user, after when updating the stream with different options, I'd like to also have the option to pick and choose to rollback to the previous version from the stream history.

Currently, I see the history showing up as:
image

It'd be nice to have the option from the list or the history page, to rollback to a pinned version. This would be an alternative to what we do via `stream rollback ... ` in the shell.

### PR DESCRIPTION
**Pull request by oodamien**
Add action to rollback to a version of a stream
Add tests
Resolves #1012

### CODE REVIEW

File: ui/src/app/streams/stream/history/stream-history.component.ts [hunk 1]
```diff
@@ -25,25 +29,79 @@ export class StreamHistoryComponent implements OnInit {
...
+            this.refresh();
+          }, error => {
+            this.notificationService.error(AppError.is(error) ? error.getMessage() : error);
+          }
+        );
+    });
+  }
+
+  /**
+   * Determine if the rollback action is allow
```
Comment (ilayaperumalg) [inline ui/src/app/streams/stream/history/stream-history.component.ts]: allow -> allowed

**Comment by ilayaperumalg** [comment]
@oodamien It would be nice if we can show the attributes of the stream in the stream history page so that the user can see which version of the stream to roll back to.

File: ui/src/app/streams/stream/history/stream-history.component.ts [hunk 2]
```diff
@@ -25,25 +29,79 @@ export class StreamHistoryComponent implements OnInit {
...
+      );
+  }
+
+  /**
+   * Rollback Action
+   * Ask a validation to the user (confirm) before performing a rollback.
+   * @param streamHistory
+   */
+  rollback(streamHistory: StreamHistory) {
+    const title = `Confirm rollback the stream`;
```
[the remark this answers is not in the document]
Comment (jvalkeal) [reply ui/src/app/streams/stream/history/stream-history.component.ts]: Maybe just `Confirm stream rollback`

File: ui/src/app/streams/stream/history/stream-history.component.ts [hunk 3]
```diff
@@ -25,25 +29,79 @@ export class StreamHistoryComponent implements OnInit {
...
+    const description = `This action will rollback the  ` +
+      `<strong>stream ${streamHistory.stream} to the version ${streamHistory.version}</strong>. Are you sure?`;
+    this.confirmService.open(title, description, { confirm: 'Rollback' }).subscribe(() => {
+      this.loggerService.log('Rollback to ' + streamHistory);
+      this.streamsService.historyRollback(streamHistory)
+        .subscribe(data => {
+            this.notificationService.success('Successfully rollback stream to the version "' + streamHistory.version + '"');
```
[the remark this answers is not in the document]
Comment (jvalkeal) [reply ui/src/app/streams/stream/history/stream-history.component.ts]: Maybe just `Successful stream rollback to version`.  Keep is short, we don't have to be so formal.

**Comment by jvalkeal** [comment]
I just directly pushed some changes to wordings using github embedded editor. Wanted to try it and it works great for small changes like this.

**Comment by jvalkeal** [comment]
ilya has a point but if that is not quick thing to do we could do it in a separate ticket?

**Comment by ilayaperumalg** [comment]
@jvalkeal yeah, I think we can track it separately. Created https://github.com/spring-cloud/spring-cloud-dataflow-ui/issues/1043 and this PR can be merged with the current updates.

**Comment by oodamien** [comment]
Thanks @jvalkeal @ilayaperumalg for the polish

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Comment by ilayaperumalg** [comment]
squashed, rebased and merged as 949eddcb

**Pull request closed without merging**